### DISCUSSION TITLE
50mb serverless function limit due to canvas module in back-end

### DISCUSSION THREAD

**Discussion by HereFrank** [vercel/community#884] [Help] [answered]
Good day to all!

I have read this discussion:
#103 
And I have read these too
https://vercel.com/docs/concepts/limits/overview#serverless-function-size

I saw all the threads, but I`m so much confused.... 
I still don't know what I should do... 

I get these errors when I try to run my project in production using vercel: 
https://i.imgur.com/EtxWbNr.png
https://i.imgur.com/mxM0ttC.png
https://i.imgur.com/6inVA4Y.png

I know that the serverless function size is too large due to the canvas module... 
But what I can do? 
I am very frustrated, please can someone give me a hand? 

I try these solution: 
https://github.com/orgs/vercel/discussions/103#discussioncomment-1769854
And these another too: 
https://github.com/prisma/prisma/issues/7184#issuecomment-1102573816

But for some reason don't work.... 
I don't know if maybe I'm doing something wrong or if is another thing...

**Comment by amyegan** [comment, marked as answer]
Hi @HereFrank! You have a couple of options that I can think of at the moment.

One is to move to to CI or deploy from local. You still wouldn't be able to keep the Canvas module in the production deployment, but you could use it during build and delete as some of the linked solutions suggest. See "[How can I use GitHub Actions with Vercel?](https://vercel.com/guides/how-can-i-use-github-actions-with-vercel)" for more info on how to set up the foundation for that approach.

Another option is to move the large package into a separate project for on-demand use. Some people have done this with things like [og-image](https://github.com/vercel/og-image) and [customized versions](https://github.com/orgs/vercel/discussions/586#discussioncomment-3181726) of it.

**Reply by HereFrank** [reply]
Thanks!
